src/number_theory/cyclotomic/rat.lean

```diff
@@ -0,0 +1,154 @@
+/-
+Copyright (c) 2022 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+
+import number_theory.cyclotomic.discriminant
+import ring_theory.polynomial.eisenstein
+
+/-!
+# Ring of integers of `p`-th power cyclotomic fields
+We compute the ring of integers of a `p ^ n`-th cyclotomic extension of `ℚ`.
+
+## Main results
+* `is_cyclotomic_extension.rat.is_integral_closure_adjoing_singleton_of_prime_pow`: if `K` is a
+  `p ^ k`-th cyclotomic extension of `ℚ`, then `(adjoin ℤ {ζ})` is the integral closure of
+  `ℤ` in `K`.
+* `is_cyclotomic_extension.rat.cyclotomic_ring_is_integral_closure_of_prime_pow`: the integral
```

Comment:
is it worth specialising these instances to `k = 1`? Not sure whether TC search will work otherwise, especially now that `x ^ 1` is not defeq to `x`...

Reply:
I've added these as separate lemmas. I don't think if they can be instances (the first one has `(hζ : ...)`) and the second one is dangerous because of `p`.